client/src/App.jsx

```diff
@@ -251,6 +272,34 @@ class App extends React.Component {
     return new Date(this.state.checkOut);
   }
 
+  showGuestPicker() {
+    this.setState({
+      guestPickerShowing: !this.state.guestPickerShowing
+    });
+  }
+
+  closeGuestPicker() {
+    this.setState({
+      guestPickerShowing: false
+    });
+  }
+
+  updateGuests(updateObj) {
+    var stateUpdateObj = {
+      numAdults: this.state.guests.numAdults,
+      numChildren: this.state.guests.numChildren,
+      numInfants: this.state.guests.numInfants
+    };
+    for (var guestType in updateObj) {
+      stateUpdateObj[guestType] = updateObj[guestType];
+    }
+    this.setState({
+      guests: stateUpdateObj
```

Comment:
Could you use the spread operator to do some of the heavy lifting here? If I understand right, you're creating a copy of `this.state.guests`, then overwriting it with any values from `updateObj`, then updating the guest state? If that's the case, you could accomplish the same with something like `this.setState({ guests: { ...this.state.guests, ...updateObj } })`